UI/UISearchBar.cs

```diff
@@ -141,6 +150,48 @@ private void HandleTextInput()
 			}
 		}
 
+		private void HandleRight()
+		{
+			if (cursorPosition != Text.Length)
+			{
+				//Check if first character is Whitespace and skip if so
+				if (char.IsWhiteSpace(Text[cursorPosition]))
+				{
+					cursorPosition += 1;
+				}
+
+				var newPos = Text.IndexOf(' ', cursorPosition);
+
+				if (newPos == -1)
+				{
+					cursorPosition = Text.Length - 1;
+				}
+				else
+				{
+					cursorPosition = newPos - 1;
+				}
+			}
+		}
+
+		private void HandleLeft()
+		{
+			if (cursorPosition > 0)
+			{
+				var newPos = Text.Substring(0, cursorPosition).TrimEnd().LastIndexOf(' ');
+
+				if (newPos == -1)
+				{
+					cursorPosition = 0;
+				}
+				else
+				{
+					cursorPosition = newPos + 1;
+				}
```

Comment:
Can this simply be `cursorPosition = newPos + 1;`?

Reply:
Yes.. I don't know why I didn't see it.